sgv2-service-common/src/main/java/io/stargate/sgv2/common/grpc/DefaultGrpcSchema.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.sgv2.common.grpc;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import io.grpc.Channel;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.grpc.stub.StreamObserver;
+import io.stargate.grpc.StargateBearerToken;
+import io.stargate.proto.QueryOuterClass.SchemaChange;
+import io.stargate.proto.Schema.CqlKeyspaceDescribe;
+import io.stargate.proto.Schema.DescribeKeyspaceQuery;
+import io.stargate.proto.Schema.GetSchemaNotificationsParams;
+import io.stargate.proto.Schema.SchemaNotification;
+import io.stargate.proto.StargateBridgeGrpc;
+import io.stargate.proto.StargateBridgeGrpc.StargateBridgeStub;
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class DefaultGrpcSchema implements GrpcSchema {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DefaultGrpcSchema.class);
+
+  private static final int KEYSPACE_CACHE_SIZE = 1000;
+  private static final Duration KEYSPACE_CACHE_TTL = Duration.ofMinutes(5);
```

Comment:
These are tentative, we can adjust them as needed, maybe make them configurable.

Reply:
Sounds good. Would be nice to somehow collect caching settings, document, but I don't know what'd be the best way. Perhaps we will just collect at a later point (easy enough to grep Caffeine Cache references)

Reply:
Also metrics